src/libcore/mem.rs

```diff
@@ -338,6 +338,34 @@ pub const fn size_of<T>() -> usize {
 /// ```
 #[inline]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[cfg(not(stage0))]
+pub const fn size_of_val<T: ?Sized>(val: &T) -> usize {
```

Comment:
If you want this PR to go through fast, you should be marking the function with `#[rustc_const_unstable(feature = "const_size_of_val")]` (and create a tracking issue)

Reply:
@oli-obk const fn features have no tracking issues for now, unfortunately. See #44644.